### PR TITLE
Tests implementation

### PR DESCRIPTION
**Pull request by sheb**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@pytest.fixture
+def config_file():
+    """
+    Create a config file providing some settings
+    and delete it at the end.
+    """
+    config = configparser.ConfigParser()
+    config["west"] = {
+        "table": "north_migrations",
+    }
```
Comment (ewjoachim) [inline tests/conftest.py]: Why do we hardcode this ? The default value is already in the CLI, no ?

File: tests/test_cli.py [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+    assert cli_runner.invoke(main, ["--version"]).output == "West 0.1.0\n"
 
 
 def test_current_database_state(cli_runner):
 
-    result = cli_runner.invoke(main, ["show_migrations"])
+    result = cli_runner.invoke(main, [
+        "--target-version",  "1.1",
+        "--migrations-root",  "example_migrations",
+        "migrate"])
```
Comment (ewjoachim) [inline tests/test_cli.py]: Don't we need to setup a specific state before the test ? Like, ensuring a fresh DB or something ?

File: tests/test_style.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    # it should does nothing
+    noop_stylist.echo("foo")
+    out, _ = capsys.readouterr()
+    assert out == ""
+
+
+@pytest.mark.parametrize("expected,checked,margin", [
+    ("  \x1b[32m[X] \x1b[0mfoo", True, 2),
+    ("  \x1b[31m[ ] \x1b[0mfoo", False, 2),
+    ("    \x1b[31m[ ] \x1b[0mfoo", False, 4),
```
Comment (ewjoachim) [inline tests/test_style.py]: Can we put these codes into variables, so we know what they mean ?

File: west/settings.py [hunk 4]
```diff
@@ -2,17 +2,27 @@
...
 logger = logging.getLogger(__name__)
 
 
-def get_config_settings():
+def get_config_settings(path):
     """
     TODO: read configuration file and return
     a dict with values to use if they are
     not overriden in env vars or CLI flags
     """
```
Comment (ewjoachim) [inline west/settings.py]: This is not a TDOD anymore :)

File: west/cli.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 CONTEXT_SETTINGS = {
     "help_option_names": ["-h", "--help"],
-    "default_map": settings.get_config_settings(),
+    "default_map": settings.get_config_settings("west.ini"),
```
Comment (ewjoachim) [inline west/cli.py]: Could be nice to load at several places, including `/etc/west.ini`, `~/.west.ini` and, yes, `./west.ini` and `./.west.ini`

Comment (k4nar) [reply west/cli.py]: Click has a helper for that: http://click.pocoo.org/6/utils/#finding-application-folders

File: tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import configparser
+import os
+import pytest
+
+
+@pytest.fixture
+def config_file():
```
Comment (k4nar) [inline tests/conftest.py]: I think this shouldn't be a fixture, but part of test_get_config_settings instead. This shouldn't be the prefered way to change settings during the tests.

File: west/cli.py (hunk 5 above)
Comment (ewjoachim) [reply west/cli.py]: Nice ! Nothing for /etc/, though, it seems.

File: tests/test_cli.py [hunk 7]
```diff
@@ -7,12 +7,28 @@ def test_version(cli_runner):
...
-    result = cli_runner.invoke(main, [
-        "--target-version",  "1.1",
-        "--migrations-root",  "example_migrations",
-        "migrate"])
+    result = cli_runner.invoke(
+        main,
+        [
+            # database connection settings
+            "--host",
+            db["host"],
```
Comment (ewjoachim) [inline tests/test_cli.py]: I'd rather we keep flags and their values on the same line.

**Comment by ewjoachim** [comment]
There are, of course, more tests to add, but we can do this in another PR.

**Comment by ewjoachim** [comment]
and great job, @sheb !

File: tests/conftest.py (hunk 6 above)
Comment (sheb) [reply tests/conftest.py]: Ok I removed the pytest fixture

**Comment by sheb** [comment]
PR rebased

**Pull request merged**